### PR TITLE
api: add pap-sha256 auth support

### PR DESCRIPTION
**Pull request by oleg-jukovec**
The authentication method works only with Tarantool EE [1].

1. https://github.com/tarantool/enterprise_doc/issues/206

I didn't forget about (remove if it is not applicable):

- [x] Tests (see [documentation](https://pkg.go.dev/testing) for a testing package)
- [x] Changelog (see [documentation](https://keepachangelog.com/en/1.0.0/) for changelog format)
- [x] Documentation (see [documentation](https://go.dev/blog/godoc) for documentation style guide)

Related issues:

Closes https://github.com/tarantool/go-tarantool/issues/243

### CODE REVIEW

File: ssl_test.go [hunk 1]
```diff
@@ -463,3 +466,44 @@ func TestSslOpts(t *testing.T) {
...
+	clientOpts := opts
+	clientOpts.Transport = "ssl"
+	clientOpts.Ssl = sslOpts
+	clientOpts.Auth = PapSha256Auth
+	conn := test_helpers.ConnectWithValidation(t, tntHost, clientOpts)
+	conn.Close()
+
+	// TODO: uncomment after the Tarantool will be available on CI:
+	// tarantool-enterprise-sdk-nogc64-2.11.0-entrypoint-110
+	// it does not work for:
```
Comment (DifferentialOrange) [inline ssl_test.go:503]: What does not work? IPROTO_ID Auth?

**Review by DifferentialOrange — APPROVED** [review]
Seems LGTM

File: ssl_test.go (hunk 1 above)
Comment (oleg-jukovec) [reply ssl_test.go:503]: Yes, it is.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ Versioning](http://semver.org/spec/v2.0.0.html) except to the first release.
 - Error type support in MessagePack (#209)
 - Event subscription support (#119)
 - Session settings support (#215)
+- pap-sha256 authorization method support (#243)
```
[the remark this answers is not in the document]
Comment (oleg-jukovec) [reply CHANGELOG.md]: Added.

**Review by DifferentialOrange — APPROVED** [review]

**Review by AnaNek — APPROVED** [review]

**Pull request merged**